internal/template-bundle/bundle.go

```diff
@@ -176,3 +191,35 @@ func createDataSource(name, namespace string) cdiv1beta1.DataSource {
 		},
 	}
 }
+
+func manageSingleReplicaInfrastructure(templates []templatev1.Template, topologyMode *osconfv1.TopologyMode, codec runtime.Encoder) error {
+	if *topologyMode != osconfv1.SingleReplicaTopologyMode {
+		return nil
+	}
+
+	for templateIdx, template := range templates {
+		for objIdx, rawObj := range template.Objects {
+			vmUnstructured, isVm, err := decodeToVMUnstructured(&rawObj)
+			if err != nil {
+				return err
+			}
+
+			if !isVm {
+				continue
+			}
+
+			val, found, err := unstructured.NestedFieldNoCopy(vmUnstructured.Object, "spec", "template", "spec", "evictionStrategy")
+			if err != nil {
+				return err
+			}
+
+			if !found || val != string(kubevirtv1.EvictionStrategyLiveMigrate) {
+				continue
+			}
+
+			unstructured.RemoveNestedField(vmUnstructured.Object, "spec", "template", "spec", "evictionStrategy")
+			templates[templateIdx].Objects[objIdx].Raw = []byte(runtime.EncodeOrDie(codec, vmUnstructured))
```

Comment:
Can you use `runtime.Encode`  and return the error, instead of `EncodeOrDie` and panic?